language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -550,21 +740,32 @@ public WorkspaceService getWorkspaceService() {
 
   @Override
   public CompletableFuture<Object> executeCommand(ExecuteCommandParams params) {
+    List<Object> args = params.getArguments();
     switch (params.getCommand()) {
       case SONARLINT_OPEN_RULE_DESCRIPTION_COMMAND:
-        List<Object> args = params.getArguments();
+        if (args == null) {
+          break;
+        }
         if (args.size() != 1) {
           warn("Expecting 1 argument");
         } else {
           String ruleKey = parseToString(args.get(0));
-          RuleDetails ruleDetails = engine.getRuleDetails(ruleKey);
+          RuleDetails ruleDetails = standaloneEngine.getRuleDetails(ruleKey);
           String ruleName = ruleDetails.getName();
           String htmlDescription = ruleDetails.getHtmlDescription();
           String type = ruleDetails.getType();
           String severity = ruleDetails.getSeverity();
           client.openRuleDescription(RuleDescription.of(ruleKey, ruleName, htmlDescription, type, severity));
         }
         break;
+      case SONARLINT_UPDATE_SERVER_STORAGE_COMMAND:
+        List<Object> list = args == null ? null : args.stream().map(SonarLintLanguageServer::parseToMap).collect(Collectors.toList());
+        handleUpdateServerStorageCommand(list);
+        break;
+      case SONARLINT_UPDATE_PROJECT_BINDING_COMMAND:
+        Map<String, Object> map = args == null || args.size() < 1 ? null : parseToMap(args.get(0));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use isEmpty() to check whether the collection is empty or not. ([squid:S1155](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1155))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=192&issues=AWMr4Qgbau6S9RpGg66z&open=AWMr4Qgbau6S9RpGg66z)

[ISSUE_KEY]: AWMr4Qgbau6S9RpGg66z